File: velolabs_repos/lattis_android/app/src/main/java/com/lattis/ellipse/data/network/base/AbstractDataMapper.java
package com.lattis.ellipse.data.network.base;

import androidx.annotation.NonNull;

import java.sql.Array;
import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;

import io.realm.RealmList;

public abstract class AbstractDataMapper<IN,OUT> {

    @NonNull
    public List<OUT> mapIn(@NonNull List<IN> ins) {
        List<OUT> outs = new ArrayList<>();
        for (IN object : ins) {
            outs.add(mapIn(object));
        }
        return outs;
    }

    @NonNull
    public abstract OUT mapIn(@NonNull IN in);

    @NonNull
    public List<OUT> mapIn(@NonNull IN[] ins) {
        return mapIn(Arrays.asList(ins));
    }

    @NonNull
    public List<IN> mapOut(@NonNull List<OUT> outs) {
        List<IN> objects = new ArrayList<>();
        for (OUT realObject : outs) {
            objects.add(mapOut(realObject));
        }
        return objects;
    }

    @NonNull
    public List<IN> mapOut(@NonNull OUT[] outs) {
        return mapOut(Arrays.asList(outs));
    }

    @NonNull
    public abstract IN mapOut(@NonNull OUT out);

}

File: velolabs_repos/lattis_android_v2/domain/src/main/java/com/lattis/domain/mapper/base/AbstractDataMapper.kt
package com.lattis.domain.mapper.base


import java.util.ArrayList



abstract class AbstractDataMapper<IN, OUT> {

    open fun mapIn(ins: List<IN>): List<OUT> {
        val outs = ArrayList<OUT>()
        for (inObjec in ins) {
            outs.add(mapIn(inObjec));
        }
        return outs
    }

    abstract fun mapIn(inObject: IN?): OUT

//    open fun mapIn(ins: Array<IN>): List<OUT> {
//        return mapIn(Arrays.asList<IN>(*ins))
//    }


    open fun mapOut(outs: List<OUT>): List<IN?> {
        val objects = ArrayList<IN?>()
        for (realObject in outs) {
            objects.add(mapOut(realObject))
        }
        return objects
    }

//    open fun mapOut(outs: Array<OUT>): List<IN> {
//        return mapOut(Arrays.asList<OUT>(*outs))
//    }

    abstract fun mapOut(out: OUT?): IN?

}
